tests/unit/test_smtp_observer.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""SMTPObserver unit tests."""
+
+# pylint: disable=protected-access
+
+import pytest
+from charms.smtp_integrator.v0.smtp import AuthType, TransportSecurity
+from ops.testing import Harness
+
+from charm_types import SMTPConfiguration
+
+RELATION_DATA_TO_SMTP_CONF_PARAMS = [
```

Comment:
Perhaps this might be better as a fixture or as a function that generates these to prevent code pollution :D

Reply:
If you agree a would leave it there, as putting it has what the tests are really checking (I tried to put it in a function, but was not improving...)

Reply:
Since there are mostly no data changes from the relation to the configuration, is this test needed?

Reply:
it tests the combinations for tls and transport_security and for auth type if plain, user and password are used. If I remove the test, these code paths will not be tested (or I will have to create several near equal tests otherwise).

Reply:
Well, IMHO this would be more in the scope of the SMTP integrator charm but I don't have a strong opinion about it.